### ISSUE TITLE
Apply cube rotation in Bonzomatic?

### ISSUE THREAD

**Issue by dacobi** [riccardoscalco/riccardoscalco.github.io#54]
Hi

I've implemented your code in Bonzomatic and have the cube seen from the front.

I am however unsure how to apply the rotation?

Edit:
I've implemented the rotate function, but can't figure out which vector(s) to rotate?
I can rotate the rayDirection, but can't rotate the cube itself.

**Comment by riccardoscalco** [comment]
Hi! Did you try to rotate both ray origin and the ray direction? Something like:

```
// Rotate the camera
vec3 axis = normalize(vec3(1.));
float angle = 1. * 3.141592653589793;
rayOrigin = rotate(rayOrigin, axis, angle);
rayDirection = rotate(rayDirection, axis, angle);
```

where the `rotate` function is the one in https://www.npmjs.com/package/glsl-rodrigues-rotation.

**Comment by dacobi** [comment]
Hi, and thanks for your reply!

I already had camera rotation working, but I've been expanding the code to render more than one object so I needed to modify the distance function to allow for rotation and translation.
However I kept getting weird bending/crushing of the cubes when applying the rotate function.

But it seems I just needed to normalize the rotation axis :)

Now my distance function looks like this:

float cubeSDF (vec3 q, vec3 mt, vec3, mr, float mangle) {
  vec3 s = vec3(0.5);
  vec3 maxis = normalize(mr);
  return length(max(abs(mrotate(q, maxis, mangle)-mt) - s, 0));
}

And as far as I can see, its working :)
